devito/data/utils.py

```diff
@@ -331,3 +309,133 @@ def mpi_index_maps(loc_idx, shape, topology, coords, comm):
         local_si[index] = as_tuple(rnorm_index)
         it.iternext()
     return owners, send, global_si, local_si
+
+
+def flip_idx(idx, decomposition):
+    """
+    This function serves two purposes:
+    1) To Convert a global index with containing a slice with step < 0 to a 'mirrored'
+       index with all slice steps > 0.
+    2) Normalize indices with slices containing negative start/stops.
+
+    Parameters
+    ----------
+    idx: Tuple of slices/ints/tuples
```

Comment:
numpydoc nitpicking:

```
idx : tuple of slices/ints/tuples
     ...
```

Reply:
Yeah, still need to fix a few docstrings.